### PR TITLE
Pricing page: Merge products and bundles section into one

### PR DESCRIPTION
**Pull request by jkguidaven**
#### Proposed Changes

* This PR will merge the Products and Bundles sections into one.
Screen Shot 2022-12-19 at 12 32 22 PM

#### Testing Instructions
1. Run this branch by following the steps below (or you can use the Jetpack Cloud live link commented on this PR)
    * Run `git fetch && git checkout update/merge-pricing-page-product-bundle`
    * Run `yarn start-jetpack-cloud`

2. Goto http://jetpack.cloud.localhost:3000/pricing or use the Jetpack cloud live link below and append `/pricing`
3. Confirm that the pricing page now has the new design

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to 1202858161751496-as-1203568228122298

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/product-store/items-list/style-most-popular.scss [hunk 1]
```diff
@@ -72,3 +72,30 @@
...
+	margin-bottom: 8px;
+
+	.amount-saved--text {
+		font-weight: 600;
+		color: var(--studio-gray-70);
+		display: flex;
+		align-items: center;
+		gap: 4px;
+	}
+
```
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/items-list/style-most-popular.scss:87]: Not related to this PR, but I see a slight misalignment of the info button for the amount saved.

Can we fix that with this bit of styling here?

```suggestion

	.amount-saved--text button {
		display: flex;
	}

```

BEFORE
Screenshot 2022-12-19 at 11 49 02 AM

AFTER
Screenshot 2022-12-19 at 11 49 53 AM

CC: @ilonagl

**Review by manzoorwanijk — APPROVED** [review]
This looks good and works as expected. I have a minor suggestion for improvement, not related to this PR.

**Comment by ilonagl** [comment]
Spacing-2
Spacing-3


@jkguidaven It looks like spacing is a bit off on mobile. I added some screenshots with spacing marks above. Please take a look and let me know your thoughts!

File: client/my-sites/plans/jetpack-plans/product-store/items-list/style-most-popular.scss (hunk 1 above)
Comment (ilonagl) [reply client/my-sites/plans/jetpack-plans/product-store/items-list/style-most-popular.scss:87]: Great catch!

**Comment by dereksmart** [comment]
Not sure if a regression - maybe with the [browser history stuff](https://github.com/Automattic/wp-calypso/pull/71300/files#diff-04270c5c1e3d5f1a470b5d0d29c71f9a99c46b83cd5e5453c8845dab62ea982fL87) being removed, or if it's a calypso.live thing: "Remove from cart" takes me back to https://wordpress.com/start/domains, rather than cloud.jetpack.com/pricing as production. 

Not a blocker for merging these changes. The rest looks good. We can address it in another PR if it's indeed regression.

**Comment by jkguidaven** [comment]
> Not sure if a regression - maybe with the [browser history stuff](https://github.com/Automattic/wp-calypso/pull/71300/files#diff-04270c5c1e3d5f1a470b5d0d29c71f9a99c46b83cd5e5453c8845dab62ea982fL87) being removed, or if it's a calypso.live thing: "Remove from cart" takes me back to https://wordpress.com/start/domains, rather than cloud.jetpack.com/pricing as production.
> 
> Not a blocker for merging these changes. The rest looks good. We can address it in another PR if it's indeed regression.


Thanks for catching this. I didn't realize this issue existed until you mentioned it. It looks like it is not related to the changes made here. I can also reproduce the issue in other PR's live links (e.g., https://container-elastic-kapitsa.calypso.live/pricing). The good thing is that the issue is not reproducible when running Calypso locally, so it is most likely an issue exclusive to the live links but I will check after merging this PR incase it resurface.

**Comment by jkguidaven** [comment]
> @jkguidaven It looks like spacing is a bit off on mobile. I added some screenshots with spacing marks above. Please take a look and let me know your thoughts!

Thanks for reviewing this. I have fixed the spacing in the mobile view as you have pointed out.

**Review by ilonagl — APPROVED** [review]

**Pull request merged**